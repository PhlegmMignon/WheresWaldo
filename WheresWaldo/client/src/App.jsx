import { useState, useEffect } from "react";
import Header from "./components/Header";
import Timer from "./components/Timer";
import Gameboard from "./components/Gameboard";
import BoardSelect from "./components/BoardSelect";
import DuringGame from "./components/DuringGame";
import PostGame from "./components/PostGame";
import BoardImages from "./BoardImages.jsx";

export default function App() {
  //DB setup

  const gameImages = BoardImages().getImages();

  useEffect(() => {
    console.log(gameImages);
  }, [gameImages]);

  const [gameState, setGameState] = useState("start");
  const [gameImage, setGameImage] = useState(gameImages[0]);
  const [found, setFound] = useState([false]);

  const setInitialFound = () => {
    let foundStatuses = [];
    gameImage.characters.forEach(() => {
      foundStatuses.push(false);
    });
    setFound(foundStatuses);
  };

  // const setFoundStatus = async (charId, clickPosition) => {
  //   if (charId < 0 || charId > found.length - 1) return false;
  //   // const valid = db.validatePosition(gameImage.id, charId, clickPos);
  //   if (!valid) {
  //     shakeGamePanel();
  //     return false;
  //   } else {
  //     let newFound = found.slice();
  //     newFound[charId] = true;
  //     setFound(newFound);
  //     checkGameWon();
  //     return true;
  //   }
  // };

  //May need to change timer later
  const renderGameState = () => {
    switch (gameState) {
      case "start":
        return (
          <BoardSelect
            images={gameImages}
            selectedImgSrc={gameImage.src}
            setGameImage={setGameImage}
            setInitialFound={setInitialFound}
            setStartTime={setStartTime}
          />
        );
      case "inProgress":
        return (
          <DuringGame
            gameImage={gameImage}
            found={found}
            // setFoundStatus={setFoundStatus}
          />
        );
      case "end":
        return <PostGame />;
    }
  };

  const toggleGameState = () => {
    switch (gameState) {
      case "start":
        setGameState("inProgress");
        break;
      case "inProgress":
        setGameState("end");
        break;
      case "end":
        setGameState("start");
    }
  };

  //Name for highscore prompt
  // const [name, setName] = useState("test");

  // //Timer states
  // const [ms, setMs] = useState("");
  const [startTime, setStartTime] = useState(0);

  // //Dropdown states
  // const [dropdownOpen, setDropdownOpen] = useState(false);
  // const [position, setPosition] = useState([0, 0]);

  return (
    <>
      <div className="App">
        <Gameboard
          gameState={gameState}
          characters={gameImage.characters}
          found={found}
        ></Gameboard>
        {renderGameState()}
      </div>
    </>
  );
}

//Character states

//Timer states
