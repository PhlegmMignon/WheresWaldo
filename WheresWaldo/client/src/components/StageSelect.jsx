// import { useEffect } from "react";

export default function StageSelect({
  images,
  setGameImage,
  setGameState,
  setInitialFound,
  setStartTime,
}) {
  const startGame = () => {
    setGameState("inProgress");
    setInitialFound();
    setStartTime(Date.now());
  };

  return (
    <div className="flex flex-col justify-center p-12  ">
      {images.map((image) => {
        return (
          <div
            key={image.id}
            className="flex flex-col items-center text-xl max-w-2xl flex-grow min-h-full"
            onClick={() => setGameImage(image)}
          >
            <img src={image.src} alt="Picture of stage" />
            <div>Convention</div>
          </div>
        );
      })}
      <button onClick={startGame}>Start</button>
    </div>
  );
}
