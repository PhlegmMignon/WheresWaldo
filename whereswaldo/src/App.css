html,
body,
#root,
.App {
  min-width: 100%;
  height: 110%;
  width: auto;
  background-color: #d3d3d3;
}

.App {
  text-align: center;
}

.login {
  display: flex;
  flex-direction: column;
  align-items: center;
}

#header {
  display: flex;
  justify-content: center;
  gap: 50px;
  align-items: end;
  height: 10%;
}

#container {
  display: flex;
  justify-content: center;
  gap: 15px;
}

.charBox {
  display: flex;
  flex-direction: column;
}

#timer1,
#timer2 {
  display: flex;
  text-align: end;
  font-size: 2rem;
}

img {
  height: 150px;
}

#main {
  height: 100%;
}
#mainImg {
  height: 100%;
  background-size: cover;
  background-image: url("./images/nMpQXwq.jpg");
}

#charDropdown {
  width: 4%;
  height: 4%;
  background-color: white;
  border-radius: 6px;
  border-top-left-radius: 0px;
}

.charDrop {
  height: 33%;
  border-bottom: 1.5px solid grey;
}

/* @media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
} */
