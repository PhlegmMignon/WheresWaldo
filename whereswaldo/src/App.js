import React, { useState, useEffect } from "react";
import { Routes, Route } from "react-router-dom";
import "./App.css";
import Home from "./components/Home";
import Start from "./components/Start";
import Main from "./components/Main";
import Header from "./components/Header";
import Timer from "./components/Timer";
import { makeTimer } from "./timerControl";
import { initializeApp } from "firebase/app";
import {
  getAuth,
  onAuthStateChanged,
  GoogleAuthProvider,
  signInWithPopup,
  signOut,
} from "firebase/auth";
import { getFirebaseConfig } from "./firebase-config.js";

export default function App() {
  const config = getFirebaseConfig();
  const app = initializeApp(config);

  //Timer states
  const [ms, setMs] = useState("");
  const [startTime, setStartTime] = useState(0);
  const [gameEnd, setGameEnd] = useState(false);

  const [gameOngoing, setGameOngoing] = useState(false);
  let checkInterval = setInterval(() => {
    if (gameOngoing) return true;
  }, 100);

  //Unmounts start when start is clicked
  useEffect(() => {
    if (gameOngoing) {
      clearInterval(checkInterval);
    }
  }, [gameOngoing]);

  //Dropdown states
  const [dropdownOpen, setDropdownOpen] = useState(false);
  const [position, setPosition] = useState([0, 0]);

  //Character found states
  const [luffyFound, setLuffyFound] = useState(false);
  const [konFound, setKonFound] = useState(false);
  const [zimFound, setZimFound] = useState(false);

  //charFound never changes in Start.js for some reason if you don't do this

  return (
    <>
      <div className="App">
        <Header
          timer={
            <Timer ms={ms} gameOngoing={gameOngoing} startTime={startTime} />
          }
        />
        <Routes>
          {/* 
        <Route
          path="/shop"
          element={
            <Shop
              setItems={setItems}
              
            />
          }
        /> */}
          <Route
            path="/"
            element={
              <Start
                setGameOngoing={setGameOngoing}
                ms={ms}
                setMs={setMs}
                setStartTime={setStartTime}
                startTime={startTime}
                luffyFound={luffyFound}
                konFound={konFound}
                zimFound={zimFound}
              />
            }
          />
          <Route
            path="/image"
            element={
              <Main
                gameOngoing={gameOngoing}
                setGameOngoing={setGameOngoing}
                setMs={setMs}
                position={position}
                setPosition={setPosition}
                dropdownOpen={dropdownOpen}
                setDropdownOpen={setDropdownOpen}
                setLuffyFound={setLuffyFound}
                setKonFound={setKonFound}
                setZimFound={setZimFound}
                luffyFound={luffyFound}
                konFound={konFound}
                zimFound={zimFound}
              />
            }
          />
        </Routes>
      </div>
    </>
  );
}

//Figure out what fxs you need to make sure firebase loads
//Add sign in btn and see if you can call the fx

//Have start btn component that loads the image
//    When last char is found, end timer is called

//Verification of character

//When a char is found, add found text under it in header
//If wrong char, have small pop up that fades after 3 secs (sticky property)

//Leaderboard info is stored in firebase.
//If not high score, show leaderboard top 5 when end timer is called and display current time
//   If high score, prompt name then show leaderboard
//When end timer is called, take current time and compare to firebase
//
