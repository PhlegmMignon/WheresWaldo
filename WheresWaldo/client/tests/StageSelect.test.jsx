import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { vi } from "vitest";
import App from "../src/App.jsx";
import StageSelect from "../src/components/StageSelect.jsx";

describe("StageSelect unit tests", () => {
  it("Renders stage select", async () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});

describe("StageSelect integration tests", () => {
  it("Clicking start button calls startGame", async () => {
    const user = userEvent.setup();
    const fakeSetGameState = vi.fn();
    const fakeSetInitialFound = vi.fn();
    const fakeSetStartTime = vi.fn();

    const fakeImages = [];
    render(
      <StageSelect
        images={fakeImages}
        setInitialFound={fakeSetInitialFound}
        setStartTime={fakeSetStartTime}
        setGameState={fakeSetGameState}
      />
    );

    const startBtn = screen.getByRole("button", { name: "Start" });
    await user.click(startBtn);

    expect(fakeSetGameState).toHaveBeenCalled();
  });
});
