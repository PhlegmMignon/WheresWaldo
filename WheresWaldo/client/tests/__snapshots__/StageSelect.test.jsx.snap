// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StageSelect unit tests > Renders stage select 1`] = `
<div>
  <div
    class="flex flex-col flex-grow "
  >
    <div
      class="bg-yellow-400 text-2xl text-center text-gray-900 p-6"
    >
      Pick a stage!
    </div>
    <div
      class="flex flex-col items-center gap-4 p-12  "
    >
      <div
        class="flex flex-col items-center text-xl max-w-2xl flex-grow min-h-full"
      >
        <img
          alt="Picture of stage"
          src="/src/assets/nMpQXwq.jpg"
        />
        <div
          class="text-lg"
        >
          convention
        </div>
      </div>
      <button
        class="text-xl"
      >
        Start
      </button>
    </div>
  </div>
</div>
`;
